app/schema/api.py

```diff
@@ -243,6 +245,38 @@ class AddPriceOut(AddPriceIn):
     id: int
 
 
+class AddDiscountIn(BaseModel):
+    """AddDiscountIn."""
+
+    vlab_id: UUID
+    discount: Decimal
+    valid_from: AwareDatetime
+    valid_to: AwareDatetime | None = None
+
+    @field_validator("discount")
+    @classmethod
+    def validate_discount(cls, v: Decimal) -> Decimal:
+        """Validate discount."""
+        if v < Decimal(0) or v > Decimal(1):
+            msg = "Discount must be between 0 and 1"
+            raise ValueError(msg)
+        return v
```

Comment:
This validation is supported by pydantic, something like:
```python
    discount: Annotated[Decimal, Field(gt=D0, lt=Decimal(1))]
```

Reply:
Implemented, I've added some tests for both ends of ranges as well.